### ISSUE TITLE
How to not lazy load Telescope ?

### ISSUE THREAD

**Issue by mandarvaze** [NvChad/NvChad#1255]
**Describe the bug**
I use `renerocksai/telekasten.nvim` plugin, which depends on `telescope.builtins`
If I start nvim, and `:Telekasten`, I get `module not found telescope.builtins`

If I use `:Telescope` first, and then use `:Telekasten`, it works without errors.

When I open `nvim` and `:PackerStatus` `telescope.nvim (not loaded)`
But after I use `:Telescope` , `not loaded` is gone from `:PackerStatus`

Based on this, it seems if I load `telescope.nvim` when nvim starts, I will be able to use `:Telekasten` upon startup.

**Expected behavior**
I should be able to use `:Telekasten` as soon as I open `nvim`
Is there a configuration to turn off lazy loading for telescope.nvim ?

**Desktop (please complete the following information):**
 - Operating System : macOS
 - Terminal: Kitty
 - Version of Neovim : 0.7.0

**Comment by siduck** [comment]
@mandarvaze in your override.lua

```lua
 ["nvim-telescope/telescope.nvim"] = {
      module = "Telescope",
   },
```
do packercompile and lemme know

**Comment by mandarvaze** [comment]
@siduck could not find `override.lua` :(
I put it in `~/.config/nvim/lua/custom/plugins/init.lua` but it did not work (obviously??)

**Comment by siduck** [comment]
you can put it in chadrc itself btw

**Comment by siduck** [comment]
```lua
M.plugins = {
   override = {
      ["nvim-telescope/telescope.nvim"] = {
         module = "Telescope",
      },
   },
}
```

**Comment by mandarvaze** [comment]
@siduck Adding 👆 in `chadrc.lua` did not help.
I tried `:PackerLoad telescope.nvim`  which seems to help.

I wonder if I can call `:PackerLoad telescope.nvim`  in `Telekasten`'s config somehow 🤔 

How ?

**Comment by mandarvaze** [comment]
Here is what worked for me (There may be better way)

```
➜ pwd
/Users/mandar/.config/nvim/lua/custom/plugins

➜ cat init.lua 

return {
  ["renerocksai/telekasten.nvim"] = {
    setup = function()
      vim.cmd("PackerLoad " .. "telescope.nvim")
    end,
  }
```

**Comment by siduck** [comment]
put it in  your custom plugins section , i totally forgot that override table is only for overriding plugin configs and not plugin packer definitions. Tested this and it works

**Comment by mandarvaze** [comment]
@siduck 

```shell
mandar in ~/.config/nvim/lua/custom on  main 
➜ cat plugins/init.lua 
return {
  ["nvim-telescope/telescope.nvim"] = {
      module = "telescope",
  },
  ["renerocksai/telekasten.nvim"] = {},
  ["simrat39/symbols-outline.nvim"] = {}
}
```

I did `:PackerSync` and restarted `nvim` 

Not working :(

**Comment by siduck** [comment]
https://user-images.githubusercontent.com/59060246/175228345-bbeccf9a-9a20-43dd-8ad8-e26985361d72.mp4

@mandarvaze

**Comment by mandarvaze** [comment]
@siduck Worked! 

Moving `telekasten` **before** `telescope` did the trick. 

![thank_you](https://user-images.githubusercontent.com/46438/175232710-ccee4352-748a-4286-8d43-4545d0167961.png)

**Comment by charludo** [comment]
Just FYI for anyone else having this problem where the solution doesn't work: this issue: wbthomason/packer.nvim#781 helped me. 

Follow the solution steps shown above, but **additionally**  add

```lua
["nvim-telescope/telescope.nvim"] = {
    cmd = "*",
},
```
to your `override.lua` or override section in `chadrc.lua`.

**Comment by siduck** [comment]
@charludo or just 

```lua
["nvim-telescope/telescope.nvim"] = {
    event = "VimEnter"
  }
  ```

**Comment by charludo** [comment]
> @charludo or just
> 
> ```lua
> ["nvim-telescope/telescope.nvim"] = {
>     event = "VimEnter"
>   }
> ```

Ah, that's better, yes.

**Comment by charludo** [comment]
Actually, I'm now running into a slightly different issue (tell me if I should rather opoen a new github issue).

I can only *either* get Telescope's builtin functionality to work, *or* Telekasten (and similar plugins).

- only adding `["nvim-telescope/telescope.nvim"] = { module = "telescope" },` to custom plugin init:
    - Telekasten works - sometimes. Try repeatedly doing `PackerSync`, close nvim, reopen. Pretty much exactly every second attempt, there will still be the error; every other time, it will work.
    - whenever it does *not* work, the builtin Telescope functions (e.g.: `<leader>ff`) *do* work; when Telekasten works, the builtin functions do *not* work
- additionally adding `["nvim-telescope/telescope.nvim"] = {  event = "VimEnter"  }` to the override section in `chadrc` results in:
    - basically the same behaviour as before, however:
    - when doing the `PackerSync`, close, reopen dance over and over again, it appears it is now more likely that the error is being thrown and the builtins work
-  also adding `cmd = { "*" }` to the override yet again drastically increases the risk of the error being thrown

I've tried a bunch of additional scenarios, but the result is always basically the same. It's pretty random whether the plugin will work, and the plugin and the builtin telescope functionality (this includes just doing `:Telescope`, btw) never work at the same time.

**Comment by siduck** [comment]
@charludo 

```lua
["telescope blabla"] = {
  rm_default_opts = true, 
  setup function,
  config function
}
```

You have to add those functions again because `rm_default_opts` will remove all of telescope's default options in its plugin definition ( which is in default plugins table )

**Comment by charludo** [comment]
Yep, placing this into custom plugin init and removing everything from the overrides worked.Thank you!!
